ui/src/features/components/JobForm/index.js

```diff
@@ -290,6 +300,14 @@ class Form extends React.Component {
         const maxVirtualUsers = parallel * 250;
         this.setState({ parallelism: parallel, max_virtual_users: maxVirtualUsers })
       }
+
+      if (this.props.webhooks.length > 0 && this.state.raw_webhooks && this.state.raw_webhooks.length !== 0 && this.state.webhooks.length === 0 && this.props.editMode) {
```

Comment:
can you explain please the purpose of this condition and its body ? 

thanks

Reply:
Hi, props.webhooks are not available right away when component is created and state.webhooks expect every item to be an option object, since it is used in dropdown. 
Back-end returns job´s webhooks in a form of id array. So this line is a trick i did to set state.webhooks once props.webhooks are received. 
Initially, received job webhook ids are stored in state raw_webhooks (which most likely should be renamed to something like webhook_ids) and then used here to map prop.webhooks to state.webhooks. `this.props.editMode` is unecessary and will be removed.
This condition is used to set state.webhooks when props.webhooks are received and state.raw_webhooks contains webhook ids.

Reply:
what do you think about making the condition in get-jobs? because the decision to show job form is there.
then in getting jobs page -  once webhooks.length > 0, find the required webhook, and show the joBform. 

thanks:) 🙏

Reply:
thats a good idea, i reworked it as you suggested, but i also added condition for a case when there are no webhooks present

Reply:
thanks!